android-lib/src/androidTest/java/com/kinvey/androidTest/store/DeltaCacheTest.java

```diff
@@ -0,0 +1,371 @@
+package com.kinvey.androidTest.store;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+import android.test.RenamingDelegatingContext;
+import android.test.suitebuilder.annotation.SmallTest;
+
+import com.kinvey.android.Client;
+import com.kinvey.android.store.DataStore;
+import com.kinvey.androidTest.TestManager;
+import com.kinvey.androidTest.callback.CustomKinveySyncCallback;
+import com.kinvey.androidTest.callback.DefaultKinveyClientCallback;
+import com.kinvey.androidTest.callback.DefaultKinveyDeleteCallback;
+import com.kinvey.androidTest.model.Person;
+import com.kinvey.java.store.StoreType;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.List;
+
+import static com.kinvey.androidTest.TestManager.PASSWORD;
+import static com.kinvey.androidTest.TestManager.USERNAME;
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+
+/**
+ * Created by yuliya on 12/27/17.
+ */
+
+@RunWith(AndroidJUnit4.class)
+@SmallTest
+public class DeltaCacheTest {
```

Comment:
Overall, these tests cover a basic set of criteria for CRUD on a single item. 
You should add a few tests on `read`, that cover reading multiple items from the backend. There should be at least one test for `read all` and one for `read by query`.

Reply:
Added